server/src/main/java/org/elasticsearch/search/fetch/subphase/FetchFieldsPhase.java

```diff
@@ -31,15 +32,6 @@ public FetchSubPhaseProcessor getProcessor(FetchContext fetchContext) {
             return null;
         }
 
-        if (fetchContext.getSearchExecutionContext().isSourceEnabled() == false) {
-            throw new IllegalArgumentException(
-                "Unable to retrieve the requested [fields] since _source is disabled "
-                    + "in the mappings for index ["
-                    + fetchContext.getIndexName()
-                    + "]"
-            );
-        }
-
```

Comment:
Yeah, it misuses it. I'm mostly ok with it because all they really want is the doc id. But there is a lot of baggage here. A cleaner interface could be much nicer.

Reply:
It'd be great to clean up this interface a bit. I don't think we had concrete ideas, would be happy to discuss in a follow-up.